homeassistant/components/light/__init__.py

```diff
@@ -287,9 +294,12 @@ async def async_handle_light_on_service(light, call):
 
             preprocess_turn_on_alternatives(hass, params)
 
-        if ATTR_PROFILE not in params:
+        if not params or not light.is_on:
             profiles.apply_default(light.entity_id, params)
 
+        if params and ATTR_TRANSITION not in params:
+            profiles.apply_default_transition(light.entity_id, params)
```

Comment:
we are now doing the lookup twice in an (almost) duplicated method. We can, alternatively, pass down the is_on to `apply_default`, and handle it just at once?

Considering your original statement in the OP of this PR:

> The documentation explicity says the default profile will be applied when the light is turned-on

This doesn't adhere to that...

Reply:
I've made this change as requested to pass through the is_on property. Should be fixed in https://github.com/home-assistant/core/pull/49376/commits/190402831f40e85f8078560bf1380f79e9486cee



> This doesn't adhere to that...
Correct. However, I have updated the documentation in https://github.com/home-assistant/home-assistant.io/pull/17513 to reflect that transition is always applied unless overwritten by the service call for a light entity. There is a use case to always apply transition. While the configuration is not exactly intuitive (hence the need for this documentation update), this new functionality is I believe intuitive.